### PR TITLE
Dense dimension sparse input

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
Do not set the output dimension of the `sparse-to-dense` layers to the same dimension as the dense features.

closes https://github.com/RasaHQ/rasa/issues/6555

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by tabergma** [comment]
/modeltest
```yml

include:
 - dataset: "Carbon Bot", "Sara","Hermit"]
   config:
      - "Sparse + DIET(bow) + ResponseSelector(bow)"
      - "Sparse + DIET(seq) + ResponseSelector(bow)"
      - "Sparse + ConveRT + DIET(bow) + ResponseSelector(bow)"
      - "Sparse + ConveRT + DIET(seq) + ResponseSelector(bow)"
      - "Sparse + Spacy + DIET(bow) + ResponseSelector(bow)"
      - "Sparse + Spacy + DIET(seq) + ResponseSelector(bow)"
```

**Comment by tabergma** [comment]
/modeltest
```yml

include:
 - dataset: 
      - "Carbon Bot"
      - "Sara"
   config:
      - "Sparse + DIET(bow) + ResponseSelector(bow)"
      - "Sparse + DIET(seq) + ResponseSelector(bow)"
      - "Sparse + ConveRT + DIET(bow) + ResponseSelector(bow)"
      - "Sparse + ConveRT + DIET(seq) + ResponseSelector(bow)"
      - "Sparse + Spacy + DIET(bow) + ResponseSelector(bow)"
      - "Sparse + Spacy + DIET(seq) + ResponseSelector(bow)"
```

**Comment by tabergma** [comment]
Results match those from the master branch (see https://metabase.rasa.com/dashboard/166)

File: rasa/nlu/selectors/response_selector.py [hunk 1]
```diff
@@ -72,6 +71,7 @@
     TENSORBOARD_LOG_LEVEL,
     CONCAT_DIMENSION,
     FEATURIZERS,
+    DENSE_DIMENSION,
```
Comment (Ghostvv) [inline rasa/nlu/selectors/response_selector.py:74]: in response selector, `128` is not enough?

Comment (Ghostvv) [reply rasa/nlu/selectors/response_selector.py:74]: since we changed it to predict labels, it is basically another DIET

Comment (tabergma) [reply rasa/nlu/selectors/response_selector.py:74]: When I set it to something lower, the performance dropped (see the tables above).

**Review by Ghostvv — APPROVED** [review]
did it reduce train run time for DIET?

**Comment by tabergma** [comment]
The train time for `Sparse + ConveRT + DIET(bow) + ResponseSelector(bow)` increased for Sara and Carbon Bot by a couple of seconds. Without ConveRT it stayed the same.

**Pull request merged**